### PR TITLE
adds crushing cans on your head and bottle flipping

### PR DESCRIPTION
**Pull request by alsoandanswer**
title.

also repaths water bottles, because it had no reason to be a can

### CODE REVIEW

**Comment by alsoandanswer** [comment]
!review

File: code/game/objects/items/trash.dm [hunk 1]
```diff
@@ -139,10 +139,15 @@
...
+	icon_state = "cola"
 	drop_sound = 'sound/items/drop/soda.ogg'
 	pickup_sound = 'sound/items/pickup/soda.ogg'
 
+/obj/item/trash/can/Initialize()
+	. = ..()
+	pixel_x = rand(-4,4)
+	pixel_y = rand(-4,4)
+
 /obj/item/trash/tuna
```
Comment (fernerr) [inline code/game/objects/items/trash.dm:151]: You can use randpixel_xy() here instead.

File: code/modules/reagents/reagent_containers/food/cans.dm [hunk 2]
```diff
@@ -4,6 +4,25 @@
...
+	desc_info = "Click it in your hand to open it.\
+				 If it's carbonated and closed, you can shake it by clicking on it with harm intent. \
+				 If it's empty, you can crush it on your forehead by selecting your head and clicking on yourself with harm intent. \
+				 You can also crush cans on other people's foreheads as well."
+
+/obj/item/reagent_containers/food/drinks/cans/attack(mob/living/M, mob/user, var/target_zone)
+	if(istype(M, /mob/living/carbon) && !reagents.total_volume && user.a_intent == I_HURT && target_zone == BP_HEAD)
+		if(M == user)
+			user.visible_message(SPAN_WARNING("[user] crushes the can of [src.name] on [user.get_pronoun(1)] forehead!"), SPAN_NOTICE("You crush the can of [src.name] on your forehead."))
+		else
```
Comment (NonQueueingMatt) [inline code/modules/reagents/reagent_containers/food/cans.dm:16]: I can already imagine the misclicks, please just make this crush the can in their hands instead LOL

Comment (alsoandanswer) [reply code/modules/reagents/reagent_containers/food/cans.dm:16]: what do you mean by misclick?

by misclick as in misinterpreting the intstructions
anyway, i added a desc_info that i felt was pretty clear, so people can execute it pretty decently.

by misclick as in, mistake, 
in any case, you'd spill the drink on yourself first anyway.
well, intentional? you need it to be empty to crush it on your head anyway.

File: code/modules/reagents/reagent_containers/food/cans.dm [hunk 3]
```diff
@@ -4,6 +4,25 @@
 	flags = 0 //starts closed
 	drop_sound = 'sound/items/drop/soda.ogg'
 	pickup_sound = 'sound/items/pickup/soda.ogg'
+	desc_info = "Click it in your hand to open it.\
+				 If it's carbonated and closed, you can shake it by clicking on it with harm intent. \
+				 If it's empty, you can crush it on your forehead by selecting your head and clicking on yourself with harm intent. \
+				 You can also crush cans on other people's foreheads as well."
+
+/obj/item/reagent_containers/food/drinks/cans/attack(mob/living/M, mob/user, var/target_zone)
+	if(istype(M, /mob/living/carbon) && !reagents.total_volume && user.a_intent == I_HURT && target_zone == BP_HEAD)
```
Comment (NonQueueingMatt) [inline code/modules/reagents/reagent_containers/food/cans.dm:13]:
```suggestion
	if(iscarbon(M)) && !reagents.total_volume && user.a_intent == I_HURT && target_zone == BP_HEAD)
```

File: code/modules/reagents/reagent_containers/food/cans.dm [hunk 4]
```diff
@@ -4,6 +4,25 @@
 	flags = 0 //starts closed
 	drop_sound = 'sound/items/drop/soda.ogg'
 	pickup_sound = 'sound/items/pickup/soda.ogg'
+	desc_info = "Click it in your hand to open it.\
+				 If it's carbonated and closed, you can shake it by clicking on it with harm intent. \
+				 If it's empty, you can crush it on your forehead by selecting your head and clicking on yourself with harm intent. \
+				 You can also crush cans on other people's foreheads as well."
+
+/obj/item/reagent_containers/food/drinks/cans/attack(mob/living/M, mob/user, var/target_zone)
+	if(iscarbon(M)) && !reagents.total_volume && user.a_intent == I_HURT && target_zone == BP_HEAD)
```
Comment (NonQueueingMatt) [inline code/modules/reagents/reagent_containers/food/cans.dm:13]:
```suggestion
	if(iscarbon(M) && !reagents.total_volume && user.a_intent == I_HURT && target_zone == BP_HEAD)
```

**Pull request merged**